### ISSUE TITLE
MainActivity is getting created twice causing app crashing in navigation

### ISSUE THREAD

**Issue by intel3** [xamarin/Xamarin.Forms#5813]
### Description
Hitting the home button and then clicking on the app icon in the home screen causes another MainActivity to create again and has weird side effect that causes a crash in the Navigation stack.

NOTE: I am only able to confirm when the apk is build from the IDE and immediately deploy to device for the first time. After the crash, relaunching the app doesn't happen again, but will appear again if you compile and deploy again.
### Steps to Reproduce
Sample project
https://1drv.ms/u/s!ApQ7q0pIhq0egVXFMuvC5OX1iZIS

1. Compile and deploy the onto Android device
2. After the app is launched hit the home button and open the app by finding the app icon in the home screen (if you reopen the app from task manager it does not happen)
3. If you put a break point in the MainActivity onCreate, you will see that it creates another activity onto the stack
4 a. At this step if you attempt to display an dialogue/alert, the app will crash with "can't attach to view" exception
4 b. After step 3 if you hit back button you will see that there is a previous created activity

However if you were to use back button at step 2, and return to the app in any mean, it will work fine as there is only one activity in the navigation stack

### Expected Behavior
It shouldn't create a new Activity on step 3
### Actual Behavior
It created a new activity and I believe xamarin form is still referencing the first activity that got created rather than the second activity that is displayed. This causing exception.
### Basic Information

- Version with issue: 3.4.0.1008975
- Last known good version: N/A
- IDE: Visual Studio for MAC 3.0 (build 3001)
- Platform Target Frameworks: 
  - Android: 9.2.0.5
- Android Support Library Version: 
- Nuget Packages: 4.8.2.5835
- Affected Devices: one plus 6T (pie) the only device I tested on

### Screenshots
N/A

### Reproduction Link
https://1drv.ms/v/s!ApQ7q0pIhq0egVTJbOmdbkZmnZLl

**Comment by kvpt** [comment]
I think the PR #4707 has fixed this issue.

Can you try if this patched version of Xamarin.Forms fix your issue ?
[xamarin.forms.3.6.3.zip](https://github.com/xamarin/Xamarin.Forms/files/3045663/xamarin.forms.3.6.3.zip)

Change the extension from .zip .to nupkg (because github limit extension type allowed).
To install the local package you can use the differents methods listed [here](https://stackoverflow.com/a/14601311).

**Comment by intel3** [comment]
I installed Xamarin.Form.3.6.3 in Android and shared project. It didn't work

**Comment by kvpt** [comment]
Ok, thank you.
So it's mean that it is another issue.
I recently worked in this area of the framework to fix similar bugs. 
I will make some tests to try to understand the issue.

**Comment by kvpt** [comment]
I don't reproduce the issue with your sample on an emulator.
For me the activity is only destroyed when using the left arrow button and no crash occur on the new activity after that.

You discribe that the issue only happen the first time and I see that you use the permission plugin.
I think it's related because the permission can cause a new activity to be created and after accepting the permission you will not be prompted to accept it on next run.

I will try some modifications to confirm my assumption.

**Comment by intel3** [comment]
You don't need permission, just build and deploy the sample app on the device, just hit home button and reopen the app via icon on home screen. Each time you do it results in a new activity getting created. You can confirm it by using the back button.
